planner/core/logical_plan_builder.go

```diff
@@ -2041,17 +2041,80 @@ func getUintFromNode(ctx sessionctx.Context, n ast.Node) (uVal uint64, isNull bo
 	return 0, false, false
 }
 
+// getUintFromLimitNode gets uint64 value from ast.Node.
+// For ordinary statement, node should be uint64 constant value.
+// For prepared statement, node is string. We should convert it to uint64.
+func getUintFromLimitNode(ctx sessionctx.Context, n ast.Node) (uVal uint64, isNull bool, isExpectedType bool) {
```

Comment:
Can we merge this function with `getUintFromNode`? They are too similar. We can add a new argument on `getUintFromNode` to indicate whether the returned value must be UINT like `getUintFromNode(ctx, n, mustUint)`.

Reply:
yeah, we can do this